test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -2439,6 +2439,21 @@ TEST_P(XdsResolverOnlyTest, KeepUsingLastDataIfBalancerGoesDown) {
   WaitForBackend(1);
 }
 
+TEST_P(XdsResolverOnlyTest, XdsStreamErrorPropagation) {
+  const std::string kErrorMessage = "test forced ADS stream failure";
+  balancer_->ads_service()->ForceADSFailure(
+      Status(StatusCode::RESOURCE_EXHAUSTED, kErrorMessage));
+  auto status = SendRpc();
+  gpr_log(GPR_INFO,
+          "XdsStreamErrorPropagation test: RPC got error: code=%d message=%s",
+          status.error_code(), status.error_message().c_str());
+  // TODO(roth): should the XDS stream error be re-packaged into an UNAVAILABLE
```

Comment:
Yes, this is definitely a bug.  I think the right place to fix this is in the xds resolver, probably right here:

https://github.com/grpc/grpc/blob/fe913387da494713a945f7842669bd3bf3c27a03/src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc#L880

Instead of just converting the `grpc_error` returned by the `XdsClient`, we should do something like this:

```
result.service_config = absl::UnavailableError(absl::StrCat(
    "error obtaining xDS resources: ", grpc_error_std_string(error)));
```

Reply:
thanks, done. Enabled check for unavailable in test